src/Services/CustomerOptionRecalculator.php

```diff
@@ -0,0 +1,62 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Brille24\SyliusCustomerOptionsPlugin\Services;
+
+use Brille24\SyliusCustomerOptionsPlugin\Entity\OrderItemInterface;
+use Brille24\SyliusCustomerOptionsPlugin\Entity\OrderItemOptionInterface;
+use Sylius\Component\Order\Factory\AdjustmentFactoryInterface;
+use Sylius\Component\Order\Model\AdjustmentInterface;
+use Sylius\Component\Order\Model\OrderInterface;
+use Sylius\Component\Order\Processor\OrderProcessorInterface;
+
+final class CustomerOptionRecalculator implements OrderProcessorInterface
+{
+    public const CUSTOMER_OPTION_ADJUSTMENT = 'CUSTOMER_OPTION_ADJUSTMENT';
+
+    /** @var AdjustmentFactoryInterface */
+    private $adjustmentFactory;
+
+    public function __construct(AdjustmentFactoryInterface $adjustmentFactory)
+    {
+        $this->adjustmentFactory = $adjustmentFactory;
+    }
+
+    public function process(OrderInterface $order): void
+    {
+        foreach ($order->getItems() as $orderItem) {
+            if (!$orderItem instanceof OrderItemInterface) {
+                continue;
+            }
+
+            $this->addOrderItemAdjustment($orderItem, $orderItem->getCustomerOptionConfiguration());
```

Comment:
Only pass the item, use the `getCustomerOptionConfiguration` in the foreach loop later.